### PR TITLE
Remove crispy::span in favor of gsl::span

### PR DESCRIPTION
**Pull request by Utkarsh-khambra**

### CODE REVIEW

**Comment by Utkarsh-khambra** [comment]
Looks like I need clang-format 14, mine clang-format keeps ruining format

File: src/crispy/span.h [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace crispy
 {
 
 template <typename T>
-class span
+class [[deprecated("Use gsl::span")]] span
```
Comment (Utkarsh-khambra) [inline src/crispy/span.h:11]: This whole file had to be reformatted to pass the style check

Comment (Utkarsh-khambra) [reply src/crispy/span.h:11]: Ok I'll remove it then

**Review by christianparpart — APPROVED** [review]
Thanks :)

**Comment by christianparpart** [comment]
Closes #747

**Pull request merged**